### DISCUSSION TITLE
embedded-postgres issue in Apple Silicon (M1)  

### DISCUSSION THREAD

**Discussion by rakeshSgr** [Sunbird-Obsrv/Community#81] [Issues] [answered]
Hi @anandp504 @manjudr 

We are facing the issue while running the test cases of sunbird-core-dataproducts in Mac m1.

Dependency

```
<dependency>
    <groupId>io.zonky.test</groupId>
    <artifactId>embedded-postgres</artifactId>
    <version>1.2.6</version>
    <scope>test</scope>
</dependency>
```

The  error we encountered : 

Exception encountered when invoking run on a nested suite - Missing postgres binaries
java.lang.IllegalStateException: Missing postgres binaries
	at io.zonky.test.db.postgres.embedded.DefaultPostgresBinaryResolver.getPgBinary(DefaultPostgresBinaryResolver.java:68)
	at io.zonky.test.db.postgres.embedded.EmbeddedPostgres.prepareBinaries(EmbeddedPostgres.java:767)
	at io.zonky.test.db.postgres.embedded.EmbeddedPostgres.<init>(EmbeddedPostgres.java:130)
	at io.zonky.test.db.postgres.embedded.EmbeddedPostgres$Builder.start(EmbeddedPostgres.java:584)
	at org.ekstep.analytics.util.EmbeddedPostgresql$.start(EmbeddedPostgresql.scala:15)
	at org.ekstep.analytics.exhaust.TestOnDemandDruidExhaustJob.beforeAll(TestOnDemandDruidExhaustJob.scala:49)
	at org.scalatest.BeforeAndAfterAll.liftedTree1$1(BeforeAndAfterAll.scala:212)

To resolve the missing binaries issue, we upgrade the embedded-postgres dependency to the latest version

```
<dependency>
    <groupId>io.zonky.test</groupId>
    <artifactId>embedded-postgres</artifactId>
    <version>2.0.4</version>
    <scope>test</scope>
</dependency>
```

After upgrading the dependency, a new error is encountered:

Exception encountered when invoking run on a nested suite - Process [/var/folders/f7/kt5z7zpj61v5b757h_wxcyh80000gn/T/embedded-pg/PG-95baeb96ab6751ea71c17b280cad95e9/bin/initdb, -A, trust, -U, postgres, -D, /var/folders/f7/kt5z7zpj61v5b757h_wxcyh80000gn/T/epg6492089817289658487, -E, UTF-8] failed
java.lang.IllegalStateException: Process [/var/folders/f7/kt5z7zpj61v5b757h_wxcyh80000gn/T/embedded-pg/PG-95baeb96ab6751ea71c17b280cad95e9/bin/initdb, -A, trust, -U, postgres, -D, /var/folders/f7/kt5z7zpj61v5b757h_wxcyh80000gn/T/epg6492089817289658487, -E, UTF-8] failed

Please help us to resolve this.

cc : @aks30 @vijiurs @Vivek-M-08

**Comment by manjudr** [comment]
There are some issues with the M1 chip and those are reported here - https://github.com/opentable/otj-pg-embedded/issues/65#issuecomment-567860383. 
Please try to export these values and run the test suit once again. 

```
export LC_ALL="en_US.UTF-8"
export LC_CTYPE="en_US.UTF-8"
```

**Comment by Vivek-M-08** [comment, marked as answer]
Hi @manjudr this issue is resolved. 

Even after updating the embedded-postgres version, IllegalStateException continued to occur. This was due to the way Mac handles the IntelliJ application, Postgres temporary files cannot be opened since access is not granted when we start the application through launchpad. In order to fix this, IntelliJ was launched by using ```idea.sh``` file in the path **cd /Applications/IntelliJ IDEA CE.app/Contents/MacOS/** via terminal. After this the test cases executed as expected.  

More about this conversation can be found here: https://github.com/zonkyio/embedded-postgres/issues/115

I happen to notice that the two lines that you shared will ensure that the local settings are correctly configured for our shell and any application that rely on these environment variables. As IntelliJ IDEA uses UTF-8 config this helped me to directly start the app through launchpad. When i added these lines in ```.zshrc``` file and refreshed with ```source .zshrc``` command.

Thank you.

**Reply by manjudr** [reply]
Great! Good to hear. Will mark it as closed.
